File: C++/53. Maximum Subarray.cpp
#include <vector>

using namespace std;

int maxSubArray(vector<int> &nums)
{
    int max_sum = nums[0], cur_sum;

    for (auto num : nums)
    {
        cur_sum += num;
        max_sum = max(max_sum, cur_sum);
        cur_sum = max(cur_sum, 0);
    }

    return max_sum;
}

File: Java/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0, maxSum = nums[0];

        for (int num : nums) {
            sum += num;
            maxSum = Integer.max(maxSum, sum);
            sum = Integer.max(sum, 0);
        }

        return maxSum;
    }
}